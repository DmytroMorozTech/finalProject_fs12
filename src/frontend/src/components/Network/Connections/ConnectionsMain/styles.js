import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  connectionsMain: {
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white
  },

  title: {
    padding: theme.spacing(2),
    paddingLeft: theme.spacing(4),
    color: theme.palette.grey[700],
    fontWeight: theme.typography.fontWeightRegular
  },

  line: {
    width: '100%',
    height: '0.5px',
    border: '0',
    margin: '0 auto',
    backgroundColor: theme.palette.grey[300]
  }
}))
