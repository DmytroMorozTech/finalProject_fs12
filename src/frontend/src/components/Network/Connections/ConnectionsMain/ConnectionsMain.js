import styles from './styles'
import Typography from '@material-ui/core/Typography'
import Connection from './Connection/Connection'
import React from 'react'

function ConnectionsMain (props) {
  const classes = styles()
  const {connections} = props

  return (
    <div className={classes.connectionsMain}>
      <Typography variant="h3" className={classes.title}>
        {connections.length} Connections
      </Typography>
      <hr className={classes.line}/>
      {connections.map(connection => <Connection key={connection.id} connection={connection}/>)}
    </div>
  )
}

export default ConnectionsMain
