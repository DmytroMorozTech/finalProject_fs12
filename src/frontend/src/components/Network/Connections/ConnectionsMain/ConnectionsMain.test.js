import React from 'react'
import ConnectionsMain from './ConnectionsMain'
import { shallow } from 'enzyme'
import * as redux from 'react-redux'

jest.mock('react-router-dom', () => ({
  Link: 'a'
}))
jest.mock('./styles', () => () => ({}))

export const connections = [
  {
    'id': 4,
    'createdDate': '2015-05-11T22:21:12.123123',
    'firstName': 'Everett',
    'lastName': 'Anderson',
    'fullName': 'Everett Anderson',
    'avatarPublicId': 'linkedin/avatars/i1bzhpqac3i1ld2urju8',
    'profileBgPublicId': '',
    'positionAndCompany': 'Project manager at HQSoftware',
    'email': 'test4@gmail.com',
    'isFollowedByActiveUser': true,
    'numberOfFollowers': 4
  },
  {
    'id': 7,
    'createdDate': '2016-03-04T09:33:31.123123',
    'firstName': 'Connie',
    'lastName': 'Johnston',
    'fullName': 'Connie Johnston',
    'avatarPublicId': 'linkedin/avatars/ob8gmrldzy35dytcib5t',
    'profileBgPublicId': '',
    'positionAndCompany': 'Full-stack developer at Amazon.com',
    'email': 'test7@gmail.com',
    'isFollowedByActiveUser': true,
    'numberOfFollowers': 1
  }
]

const setUp = (props) => shallow(<ConnectionsMain {...props}/>)

describe('should render ConnectionsMain component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component
  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp({connections})
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
