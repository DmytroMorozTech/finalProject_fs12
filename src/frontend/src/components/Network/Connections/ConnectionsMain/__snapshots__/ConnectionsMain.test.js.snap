// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConnectionsMain component to match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Typography))
    variant="h3"
  >
    2
     Connections
  </WithStyles(ForwardRef(Typography))>
  <hr />
  <Connection
    connection={
      Object {
        "avatarPublicId": "linkedin/avatars/i1bzhpqac3i1ld2urju8",
        "createdDate": "2015-05-11T22:21:12.123123",
        "email": "test4@gmail.com",
        "firstName": "Everett",
        "fullName": "Everett Anderson",
        "id": 4,
        "isFollowedByActiveUser": true,
        "lastName": "Anderson",
        "numberOfFollowers": 4,
        "positionAndCompany": "Project manager at HQSoftware",
        "profileBgPublicId": "",
      }
    }
    key="4"
  />
  <Connection
    connection={
      Object {
        "avatarPublicId": "linkedin/avatars/ob8gmrldzy35dytcib5t",
        "createdDate": "2016-03-04T09:33:31.123123",
        "email": "test7@gmail.com",
        "firstName": "Connie",
        "fullName": "Connie Johnston",
        "id": 7,
        "isFollowedByActiveUser": true,
        "lastName": "Johnston",
        "numberOfFollowers": 1,
        "positionAndCompany": "Full-stack developer at Amazon.com",
        "profileBgPublicId": "",
      }
    }
    key="7"
  />
</div>
`;
